src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -716,23 +716,25 @@ private Vector(ref Register existingRegister)
         /// <summary>
         /// Constructs a vector from the given <see cref="ReadOnlySpan{Byte}"/>. The span must contain at least <see cref="Vector{Byte}.Count"/> elements.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Vector(ReadOnlySpan<byte> values)
             : this()
         {
             ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
-            if (!MemoryMarshal.TryRead(values, out this))
+            if (values.Length < Vector<byte>.Count)
             {
                 Vector.ThrowInsufficientNumberOfElementsException(Vector<byte>.Count);
             }
+            this = Unsafe.ReadUnaligned<Vector<T>>(ref MemoryMarshal.GetReference(values));
         }
         
         /// <summary>
         /// Constructs a vector from the given <see cref="ReadOnlySpan{T}"/>. The span must contain at least <see cref="Vector{T}.Count"/> elements.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Vector(ReadOnlySpan<T> values)
             : this()
         {
-            ThrowHelper.ThrowForUnsupportedVectorBaseType<T>();
```

Comment:
Why was this removed? This is a public method...

Reply:
Not needed in methods which access the `Count` property, as the property getter would throw. (This pattern is used elsewhere in this file.)

Reply:
Thanks, I missed that bit.